geoh5py/objects/geo_image.py

```diff
@@ -371,15 +372,16 @@ def image(self, image: str | np.ndarray | BytesIO | Image.Image):
                 raise ValueError(f"Input image file {image} does not exist.")
 
             image = Image.open(image)
-
             # if the image is a tiff save tag information
             if isinstance(image, TiffImageFile):
                 self.tag = image
+                convert_to_grid = True
```

Comment:
Combine duplicate logic. All of these could be just if statements that converts nd.array or str or bytes to an image. Then do the checks for is `Image` and is `TiffImageFile` once